btm/src/test/java/bitronix/tm/mock/NewJdbcProperUsageMockTest.java

```diff
@@ -1156,6 +1245,29 @@ public void testAutoCommitTrueWhenEnlistedButSuspended() throws Exception {
         tm.shutdown();
     }
 
+    public void testAutoCommitTrueWhenEnlistedButQuickSuspended() throws Exception {
+       Thread.currentThread().setName("testAutoCommitTrueWhenEnlistedButQuickSuspended");
+       TransactionManagerServices.getConfiguration().setQuickSuspend(true);
+       BitronixTransactionManager tm = TransactionManagerServices.getTransactionManager();
+
+       tm.begin();
+
+       Connection c = poolingDataSource1.getConnection();
+       c.prepareStatement("");
+
+       Transaction tx = tm.suspend();
+       assertNull(tm.getTransaction());
+
+       assertTrue(c.getAutoCommit());
```

Comment:
I think I see what you are saying, but let me make sure.

The reason that getAutoCommit() is true, is because the call to jdbcPooledConnection.isParticipatingInActiveGlobalTransaction() returns false.  That code already reported false when suspended was true (though before my change, that was always associated with end).

As of now, since there is no currentTransaction, this method always returns false.  I suppose you are thinking that even though we're suspended, we're effectively in an ActiveGlobalTransaction.  If you look at the code that uses the isParticipatingIn..  call that makes sense to me.

So, I might need a bit of help if we were going to try to return true in this state.  Since getCurrentTransaction is null - we don't know the exact gtrid to answer with.  Should we evaluate all of the xaResourceHolderStates?  Is there a better way to determine if we're started, not ended, and suspended?

There is a confusing comment on isParticipatingIn...  call.  It says  "@return true if start() has been successfully called but not end() yet <i>and</i> the transaction is not suspended.".  We should update that to reflect the final direction.